packages/flutter/lib/src/widgets/value_listenable_builder.dart

```diff
@@ -86,6 +87,18 @@ class ValueListenableBuilder<T> extends StatefulWidget {
   /// null, therefore your [builder] may need to handle null values.
   final ValueListenable<T> valueListenable;
 
+  /// Whether the [builder] should be called only when the [valueListenable]'s
+  /// value changes.
+  ///
+  /// If `true`, the [builder] is called only when the [valueListenable]'s value
+  /// changes, reusing the previously built widget for other rebuilds (e.g., due
+  /// to parent widget updates). If `false` (default), the [builder] is called
+  /// on every build, reflecting the current value of the [valueListenable].
+  ///
+  /// Setting this to `true` can improve performance by reducing unnecessary
+  /// rebuilds when the [valueListenable]'s value hasn't changed.
+  final bool rebuildOnlyOnValueChange;
```

Comment:
I'd rather see something like this `bool Function(T oldValue, T newValue)? buildWhen`.

Then, if provided, you can filter rebuilds in a more flexible way (I.e. if only one property in the value changes)
Also, instead of caching an old widget/managing a separate `_isChangeCalled`, just pevent the call to `setState` if buildWhen returns false? (but still set the `value` in _ValueListenableBuilderState)

Reply:
But without caching, the `builder` still rebuilds whenever the parent widget's state changes, even if the `ValueListenable` value remains unchanged! How to prevent this without caching ?

Reply:
Hmm, so expecting developers to manually optimize the parent to prevent `ValueListenableBuilder` rebuilds can be cumbersome, especially in complex widget trees or with third-party widget. Adding `rebuildOnlyOnValueChange` would simplify this for developers, letting them focus on the `ValueListenable`’s logic.